src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -59,4 +59,182 @@ private bool IsMainWindow(IntPtr handle)
+                                if (enumResult)
+                                {
+                                    break;
+                                }
+                                Thread.Sleep(1);
+                            }
+                        }
+                    }
+                    finally
+                    {
+                        moduleHandlesArrayHandle.Free();
+                    }
+
+                    if (!enumResult)
+                    {
+                        throw new Win32Exception();
+                    }
+
+                    moduleCount /= IntPtr.Size;
+                    if (moduleCount <= moduleHandles.Length)
+                        break;
+                    moduleHandles = new IntPtr[moduleHandles.Length * 2];
+                }
+
+                var modules = new ProcessModuleCollection(firstModuleOnly ? 1 : moduleCount);
+
+                char[] chars = new char[1024];
+
+                for (int i = 0; i < moduleCount; i++)
+                {
+                    if (i > 0)
+                    {
+                        // If the user is only interested in the main module, break now.
+                        // This avoid some waste of time. In addition, if the application unloads a DLL
+                        // we will not get an exception.
+                        if (firstModuleOnly)
+                        {
+                            break;
+                        }
+                    }
+
+                    IntPtr moduleHandle = moduleHandles[i];
+                    Interop.Kernel32.NtModuleInfo ntModuleInfo;
+                    if (!Interop.Kernel32.GetModuleInformation(processHandle, moduleHandle, out ntModuleInfo))
+                    {
+                        HandleError();
+                        continue;
+                    }
+
+                    var module = new ProcessModule()
+                    {
+                        ModuleMemorySize = ntModuleInfo.SizeOfImage,
+                        EntryPointAddress = ntModuleInfo.EntryPoint,
+                        BaseAddress = ntModuleInfo.BaseOfDll
+                    };
+
+                    int length = Interop.Kernel32.GetModuleBaseName(processHandle, moduleHandle, chars, chars.Length);
+                    if (length == 0)
+                    {
+                        HandleError();
+                        continue;
+                    }
+
+                    module.ModuleName = new string(chars, 0, length);
+
+                    length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                    if (length == 0)
+                    {
+                        HandleError();
+                        continue;
+                    }
+
+                    module.FileName = (length >= 4 && chars[0] == '\\' && chars[1] == '\\' && chars[2] == '?' && chars[3] == '\\') ?
```

Comment:
This isn't catching all device paths. Additionally this would be giving back unusable paths in some cases. Stripping here isn't a good idea.